chart/templates/dashboards/openstack-dashboard.yaml

```diff
@@ -0,0 +1,6871 @@
+#
+# Copyright 2020 The dNation Kubernetes Monitoring Authors. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+{{- if and .Values.grafanaDashboards.enable .Values.openstackMonitoring.enabled }}
+{{- $labelKey := first (keys .Values.grafanaDashboards.labelGrafana) }}
+{{- $labelValue := first (values .Values.grafanaDashboards.labelGrafana) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: {{ include "k8s-monitoring.namespace" . }}
+  name: {{ include "k8s-monitoring.fullname" . }}-openstack-dashboard
+  labels:
+    {{ include "k8s-monitoring.labels" . | indent 4 }}
+    app: {{ include "k8s-monitoring.name" . }}
+    {{ $labelKey }}: {{ $labelValue | quote }}
+    release: {{ include "k8s-monitoring.release" . }}
+  annotations:
+    {{ include "k8s-monitoring.annotations" . | indent 4 }}
+    {{- if .Values.grafanaDashboards.destinationOverride }}
+    {{ toYaml .Values.grafanaDashboards.destinationOverride | indent 4 }}
+    {{- end }}
+data:
+  openstack.json: |-
+    {
+    "annotations": {
+        "list": [
+            {
+                "builtIn": 1,
+                "datasource": {
+                    "type": "grafana",
+                    "uid": "-- Grafana --"
+                },
+                "enable": true,
+                "hide": true,
+                "iconColor": "rgba(0, 211, 255, 1)",
+                "name": "Annotations & Alerts",
+                "type": "dashboard"
+            }
+        ]
+    },
+    "description": "OpenStack dashboard using the Prometheus OpenStack exporter (https://github.com/openstack-exporter/openstack-exporter)",
+    "editable": true,
+    "fiscalYearStartMonth": 0,
+    "graphTooltip": 0,
+    "id": 3568,
+    "links": [],
+    "liveNow": false,
+    "panels": [
+        {
+            "collapsed": false,
+            "gridPos": {
+                "h": 1,
+                "w": 24,
+                "x": 0,
+                "y": 0
+            },
+            "id": 15,
+            "panels": [],
+            "title": "Service status",
+            "type": "row"
+        },
+        {
+            "datasource": {
+                "type": "prometheus",
+                "uid": "PC96415006F908B67"
```

Comment:
IMHO, this is not good because different deployments of Grafana don’t necessarily have the same UID. Check also other datasources please.